features/embedded-wallets/turnkey-embedded-wallet.ts

```diff
@@ -0,0 +1,52 @@
+import { onPasskeyCreate } from "@/utils/passkeys/create-passkey";
+import { IEmbeddedWallet } from "./embedded-wallet.interface";
+import { Client } from "@xmtp/react-native-sdk";
+import { LocalAccount } from "viem/accounts";
+import {
+  copyDatabasesToTemporaryDirectory,
+  createTemporaryDirectory,
+} from "@/utils/fileSystem";
+import { getDbEncryptionKey } from "@/utils/keychain/helpers";
+import { getInboxId } from "@/utils/xmtpRN/signIn";
+import { viemSignerToXmtpSigner } from "@/utils/xmtpRN/signer";
+
+export class TurnkeyEmbeddedWallet implements IEmbeddedWallet {
+  client: LocalAccount | null = null;
+  xmtpClient: Client | null = null;
+
+  constructor(public readonly address: string) {}
+
+  async createClient() {
+    const client = await onPasskeyCreate();
+    if (!client) {
+      throw new Error("Failed to create client");
+    }
+    return client;
+  }
+
+  async loadClient() {
+    // return new LocalAccount(this.address);
+  }
+
+  async instantiateXmtpClient() {
+    if (!this.client) {
+      throw new Error("Client not found");
```

Comment:
This feels wrong if we’re extension in a client why are we throwing? Is the clients not found?
We shouldn’t expect a client if we’re in instantiating I think we just need to remove this bam  or add one more